src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs

```diff
@@ -131,25 +131,36 @@ public override void VisitSwitchCase(ISwitchCaseOperation operation)
             Assert.Equal(OperationKind.SwitchCase, operation.Kind);
             VisitLocals(operation.Locals);
             AssertEx.Equal(operation.Clauses.Concat(operation.Body), operation.Children);
+            IOperation condition = ((BaseSwitchCase)operation).Condition;
+
+            if (condition != null)
+            {
+                var explictNodeMap = new Dictionary<SyntaxNode, IOperation>();
+
+                foreach (IOperation descendant in condition.DescendantsAndSelf())
+                {
+                    if (!descendant.IsImplicit)
+                    {
+                        try
+                        {
+                            explictNodeMap.Add(descendant.Syntax, descendant);
+                        }
+                        catch (ArgumentException)
+                        {
+                            Assert.False(true, $"Duplicate explicit node for syntax ({descendant.Syntax.RawKind}): {descendant.Syntax.ToString()}");
```

Comment:
Consider using `Assert.False(explicitNodeMap.ContainsKey(descendant.Syntax), $"Duplicate ...")` rather than `try/catch`.

Reply:
> Consider using Assert.False(explicitNodeMap.ContainsKey(descendant.Syntax), $"Duplicate ...") rather than try/catch.

I need to add key to the dictionary as well. Killing two birds with one stone.

---
In reply to: [184550508](https://github.com/dotnet/roslyn/pull/26395#discussion_r184550508) [](ancestors = 184550508)